opentelemetry-csharp-sdk-to-seq/Example.WeatherService/Program.cs

```diff
@@ -3,74 +3,60 @@
-                    ["deployment.environment"] = "development"  
-                }));
-  
-        // Some important options to improve data quality
-        openTelemetryLoggerOptions.IncludeScopes = true;
-        openTelemetryLoggerOptions.IncludeFormattedMessage = true;
+var builder = WebApplication.CreateBuilder(args);
 
-        openTelemetryLoggerOptions.AddConsoleExporter();
-        openTelemetryLoggerOptions.AddOtlpExporter(opt =>
+builder.Services.AddOpenTelemetry()
+    .ConfigureResource(resource => resource.AddService(serviceName: builder.Environment.ApplicationName))
+    .WithTracing(tracing =>
+    {
+        tracing.AddSource(exampleActivitySource.Name);
+        tracing.AddAspNetCoreInstrumentation();
+        tracing.AddConsoleExporter();
+        tracing.AddOtlpExporter(opt =>
         {
-            opt.Endpoint = new Uri("http://localhost:5341/ingest/otlp/v1/logs");
+            opt.Endpoint = new Uri("http://localhost:5341/ingest/otlp/v1/traces");
             opt.Protocol = OtlpExportProtocol.HttpProtobuf;
             //opt.Headers = "X-Seq-ApiKey=abcde12345";
         });
-    }));
+    });
 
-    var app = builder.Build();
-    app.Logger.LogInformation("Starting {App}", builder.Environment.ApplicationName);
+builder.Services.AddLogging(logging => logging.AddOpenTelemetry(openTelemetryLoggerOptions =>  
+{  
+    openTelemetryLoggerOptions.SetResourceBuilder(  
+        ResourceBuilder.CreateEmpty()
+            .AddService(serviceName: builder.Environment.ApplicationName));
 
-    var forecastByPostcode = Directory.GetFiles("./data")
-        .ToDictionary(f => Path.GetFileNameWithoutExtension(f)!, f => File.ReadAllText(f).Trim());
-    
-    app.MapGet("/{postcode}", (string postcode) =>
+    // Some important options to improve data quality
+    openTelemetryLoggerOptions.IncludeScopes = true;
+
+    openTelemetryLoggerOptions.AddConsoleExporter();
+    openTelemetryLoggerOptions.AddOtlpExporter(opt =>
     {
-        using var activity = ExampleWeatherServiceActivitySource.StartActivity("Look up forecast for postcode {Postcode}");
-        activity?.SetTag("Postcode", postcode);
-        var forecast = forecastByPostcode[postcode];
-        activity?.SetTag("Forecast", forecast);
-        
-        return forecast;
+        opt.Endpoint = new Uri("http://localhost:5341/ingest/otlp/v1/logs");
+        opt.Protocol = OtlpExportProtocol.HttpProtobuf;
+        //opt.Headers = "X-Seq-ApiKey=abcde12345";
     });
+}));
 
-    app.Run();
+var app = builder.Build();
+app.Logger.LogInformation("Starting {App}", builder.Environment.ApplicationName);
 
-    return 0;
-}
-catch (Exception)
+var forecastByPostcode = Directory.GetFiles("./data")
+    .ToDictionary(f => Path.GetFileNameWithoutExtension(f)!, f => File.ReadAllText(f).Trim());
+
+app.MapGet("/{postcode}", (string postcode) =>
 {
-    return 1;
-}
+    using var activity = exampleActivitySource.StartActivity("Look up forecast for postcode {Postcode}");
```

Comment:
The suppressed warning at the top of the file is because we want to pass an activity name explicitly here.